mypy/plugins/dataclasses.py

```diff
@@ -469,9 +469,15 @@ def add_slots(
                 self._cls,
             )
             return
-
         info.slots = generated_slots
 
+        # Now, insert `.__slots__` attribute to class namespace:
+        slots_type = TupleType(
+            [self._api.named_type("builtins.str") for _ in generated_slots],
```

Comment:
Is it worth creating Literal types here? So the inferred type in the test cases would be `tuple[Literal["x"], Literal["y"], Literal["z"]]`.

Reply:
No, because regular classes produce the same result: 

```python
class A:
    __slots__ = ('a', 'b')

reveal_type(A.__slots__)  # N: Revealed type is "tuple[builtins.str, builtins.str]"
```

`__match_args__` attribute needs `Literal` to handle match kwargs correctly, but `__slots__` does not need it at all.

Reply:
Thanks, I feel like we might get an issue at some point asking that we use the literal types, but it's fine if we're consistent with how explicit `__slots__` works.